### DISCUSSION TITLE
How to do a wildcard proxy with 3.0?

### DISCUSSION THREAD

**Discussion by SantoJambit** [chimurai/http-proxy-middleware#1003] [Q&A] [answered]
With 2.x I had the following code:

```ts
app.use(
            "*",
            createProxyMiddleware({
                target: "http://localhost:4200",
                changeOrigin: true,
            })
        );
```

This would forward all unknown requests (not already handled by express) to a separate server. In 3.0, this always responds with the index page of http://localhost:4200, rather than the specific sub-page that was actually requested.

I don't see how I can do this with 3.0. Nothing in the migration guide seems to point to anything like it.

Thanks in advance

**Comment by leoc-22** [comment, marked as answer]
just encountered this issue this week and my solution is to use `pathRewrite` to return `path`. 

in my use case, it was to proxy anything with `/api/*` to the target url:

```typescript
app.use(
  "/api",
  createProxyMiddleware({
      pathRewrite: (path) => path
      target: "http://localhost:4200",
      changeOrigin: true,
  })
);
```

you can probably also use:

```typescript
on: {
  proxyReq: (proxyReq, req, res) => {}
}
```

to get the path after the base url and attach to the target base url.

**Reply by SantoJambit** [reply]
Thanks for the reply. At first it did not work for me.. pathRewrite would receive a `"/"` no matter the path.
But looking at your code, you didn't use an asterisk, so I changed the first parameter of `app.use` to "/" and it seems to do the trick:

```ts
        app.use(
            "/",
            createProxyMiddleware({
                pathRewrite: (path) => path,
                target: "http://localhost:4200",
                changeOrigin: true,
            })
        );
```
